# HG changeset patch
# User David Bateman <dbateman@free.fr>
# Date 1283542511 -7200
#      Fri Sep 03 21:35:11 2010 +0200
# Node ID 90450db218e46e780128980023bd7c3016aa8979
# Parent  48a1812fcc3a433a1b588c2bc61f232fc507b8d1
Allow the delaunay function to treat matrices of the same size for compatibility

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2010-07-07  David Bateman  <dbateman@free.fr>
+
+	* geometry/delaunay.m: Allow the delaunay function to treat
+	matrices of the same size for compatibility.
+
 2010-09-03  Ben Abbott <bpabbott@mac.com>
 
 	* image/image.m: Matlab compatible axis limits, add demo.
 
 2010-09-01  Jaroslav Hajek  <highegg@gmail.com>
 
 	* io/dlmwrite.m: Don't fclose if file ID is supplied.
 
diff --git a/scripts/geometry/delaunay.m b/scripts/geometry/delaunay.m
--- a/scripts/geometry/delaunay.m
+++ b/scripts/geometry/delaunay.m
@@ -45,26 +45,27 @@
 ## Author: Kai Habel <kai.habel@gmx.de>
 
 function ret = delaunay (x, y, opt)
 
   if (nargin != 2 && nargin != 3)
     print_usage ();
   endif
   
-  if (isvector (x) && isvector (y) && length (x) == length (y))
+  if ((isvector (x) && isvector (y) && length (x) == length (y))
+      || size_equal (x, y))
     if (nargin == 2)
       tri = delaunayn ([x(:), y(:)]);
     elseif (ischar (opt) || iscellstr (opt))
       tri = delaunayn ([x(:), y(:)], opt);
     else
       error ("delaunay: third argument must be a string");
     endif
   else
-    error ("delaunay: first two input arguments must be vectors of same size");
+    error ("delaunay: first two input arguments must be matrices of same size");
   endif
 
   if (nargout == 0)
     x = x(:).';
     y = y(:).';
     X = [x(tri(:,1)); x(tri(:,2)); x(tri(:,3)); x(tri(:,1))];
     Y = [y(tri(:,1)); y(tri(:,2)); y(tri(:,3)); y(tri(:,1))];
     plot(X, Y, 'b', x, y, 'r*');
